gcsfs/tests/test_core.py

```diff
@@ -1053,7 +1053,10 @@ def test_placeholder_dir_cache_validity():
         gcs.touch(f"gs://{TEST_BUCKET}/a/file")
         gcs.touch(f"gs://{TEST_BUCKET}/b/file")
 
-        # Confirm the lookup works before the cache is polluted
-        gcs.info(f"gs://{TEST_BUCKET}/b/file")
+        gcs.ls(f"gs://{TEST_BUCKET}/", prefix="a/")
```

Comment:
The purpose of prefix in `ls` is to subselect paths within a listing that is *not* a directory level, i.e., 
`prefix="path/directory/file"` to select "path/directory/file*" and similar without listing the whole of "directory". Perhaps this is the problem? I don't think that a `prefix` value ending in "/" makes sense.

Reply:
Yeah the trailing `/` case may be a bit degenerate, but I updated it to use `a/file` to match `file1` and `file2` and the same issue arises.